source/features/fault.cc

```diff
@@ -488,8 +490,25 @@ namespace WorldBuilder
       // todo: explain
       const double starting_radius = natural_coordinate.get_depth_coordinate() + depth - starting_depth;
 
+      double buffer_zone = 0.;
+
+      // Here, we check if all the coordinate points describing a fault lie within a bounding box. To do this, we use min and
+      // max points for a fault and add a buffer zone surrounding it. Since the fault extends over a depth (i.e., fault lenght),
+      // the idea is to use a small buffer zone using the approximate length of fault, such that the box covers the fault
+      // extent in depth. For spherical system, we assume that the equivalent buffer is spherical distance computed at the equator.
+
+      if (world->parameters.coordinate_system->natural_coordinate_system() == CoordinateSystem::spherical)
+        buffer_zone = 2 * (maximum_fault_thickness + maximum_total_fault_length) / (const_pi * starting_radius* 2);
+      else if (world->parameters.coordinate_system->natural_coordinate_system() == CoordinateSystem::cartesian)
+        buffer_zone = 2 * (maximum_fault_thickness + maximum_total_fault_length);
+
+      std::vector<Point<2> >  bounding_box_coordinates = WorldBuilder::Utilities::get_bounding_box (
+                                                           reference_point, coordinates, original_number_of_coordinates, buffer_zone);
```

Comment:
This is something that doesn't depend on the current point this function is checking. It would be easiest if you could just do this computation once after you read in the details of the fault in `parse_entries()`, and then store the bounding box in a member variable.

Reply:
I remember moving the calculation of ```bounding_box_coordinates()``` in ```parse_entries()``` stage based on @MFraters 's suggestion. However, the reason I moved it back into the ```get_composition``` stage is that I needed to use the ```starting_radius``` value for computing the buffer_zone, which I couldn't use in  ```parse_entries()```. Maybe there is a way to resolve this by modifying the approach for ```buffer_zone``` computation.

Reply:
I just realized that I can access the ```starting_radius``` value in the ```parse_entries()```. Please ignore my previous comment.

Reply:
The `starting_radius` is dependent on both the position and depth of the request. You only have them during the request itself (i.e. `get_composition`). I don't see how you could get those values during the `parse_entries` phase.

Reply:
You are right! I am sorry, I confused myself by using ```starting_depth```.

Reply:
Is that because you are building 2d bounding boxes only for the specific depth you are currently at?

Reply:
No, it is general for all cases. I tried to explain the function in a comment reply below.